coral-spark/src/main/java/com/linkedin/coral/spark/dialect/SparkSqlDialect.java

```diff
@@ -30,8 +31,12 @@
  */
 public class SparkSqlDialect extends SqlDialect {
 
-  public static final SparkSqlDialect INSTANCE = new SparkSqlDialect(
-      emptyContext().withDatabaseProduct(DatabaseProduct.HIVE).withNullCollation(NullCollation.HIGH));
+  public static final SqlDialect.Context DEFAULT_CONTEXT =
+      SqlDialect.EMPTY_CONTEXT.withDatabaseProduct(DatabaseProduct.SPARK).withLiteralQuoteString("'")
+          .withLiteralEscapedQuoteString("\\'").withNullCollation(NullCollation.LOW)
+          .withUnquotedCasing(Casing.UNCHANGED).withQuotedCasing(Casing.UNCHANGED).withCaseSensitive(false);
```

Comment:
In previous impl, we are using `NullCollation.HIGH`,  `Casing.TO_UPPER` for `withUnquotedCasing` and `true` for `withCaseSensitive`, why are we changing here?

Reply:
I think they are more correct for spark-sql? And seems there's no regressions for this change so I think it's fine to add them?

see ref: https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java#L47

https://community.databricks.com/t5/data-engineering/spark-is-case-sensitive-spark-is-not-case-sensitive-by-default/td-p/13954#:~:text=is%20case%20sensitive%3F-,Spark%20is%20not%20case%20sensitive%20by%20default.,will%20get%20column%20ambiguity%20error.

Reply:
I think these configs are mostly used when using calcite parser to parser a sql dialect, we are not using them here, so basically they are not effective at all.

Maybe later when we want to use calcite to parse spark-sql in coral, they can become useful.